src/block/BlockManager.ts

```diff
@@ -82,29 +86,95 @@ export default class BlockManager {
     }
 
     private async generateBlockPalette() {
-        let palette: BinaryStream = await new Promise((resolve) => {
-            let data: BinaryStream = new BinaryStream();
-            let writer: NBTWriter = new NBTWriter(
-                data,
-                ByteOrder.LITTLE_ENDIAN
-            );
-            writer.setUseVarint(true);
+        this.server.getLogger().debug('Checking block palette...');
 
-            this.getBlocks()
-                .filter((b) => b.meta === 0)
-                .map((block) => () => {
-                    let tag: NBTTagCompound = new NBTTagCompound('');
-                    let nbtBlock: NBTTagCompound = new NBTTagCompound('block');
+        try {
+            this.blockPalette = await this.server
+                .getCacheManager()
+                .getCachedPalette();
```

Comment:
We need to handle updating the change when the blockdata changes. (Eg new versions or a plugin author registers one).

Reply:
Plugins shouldn't register any block palette. Versioned cache commit is almost finished 👍

Reply:
@HerryYT They might however register a new block. Which will require a new block palette generation.

Reply:
E.g. use something like this for the version id `sha256(Array.from(this.blocks).map(a => a.name).sort((a, b) => a - b).join() + prismarine_version)`

Reply:
That way if `this.blocks` changes we'll automatically generate a new palette.